### ISSUE TITLE
New tab will open scrolling tab bar all the way to the left.

### ISSUE THREAD

**Issue by ttggzz** [ssborbis/ContextSearch-web-ext#271]
Hello, not sure if the is FFx problem or an add-on but I've been noticing it for some time now. I use both this add-on and "open tabs next to current". I regularly have large tab counts in multiple windows. Let's say I have 200 tabs open and am working from tabs in the middle of the tab toolbar. When opening a new tab via ContextSearch-web-ext it will quite often open at the extreme left, jerking the entire viewable tabs over there with it. I then need to manually scroll the tabs back to the middle. 

The new tab opened via a ContextSearch will indeed be next to the last current tab, but like I said everything gets scrolled over to the extreme left. Sometimes middle clicking the new search works, the tabs will not change position in the toolbar, but not always. Any way to stop this? thank you

**Comment by ssborbis** [comment]
I'm assuming you're using the default settings, but just to be sure, what mouse button are you using to search? What do you have your mouse buttons set to do in this add-on when performing a search? Open to New Tab? Open In Background Tab? Are you using the context menu or the quick menu?

**Comment by ssborbis** [comment]
I'm browsing the code for "open tabs next to current (OTNC)" and I'm pretty sure that's the issue. This add-on (CS) uses very basic code when opening new tabs. That other addon tracks new tab creation and moves them based on a running list of tab ids. It's expecting new tabs to respect it's own tab management and sorting when the ` browser.tabs.onCreated` event fires. Well, I need that event too and I set the parent tab as the tab you perform the search from, as should be the case. That must confuse OTNC which isn't accounting for other addons using the tab creation api

I could add an advanced option to ignore explicitly setting the parent tab id, but you'd need to do the testing.

**Comment by ttggzz** [comment]
Thanks for the quick response. Yes I believe I am using the default settings, context menu. Left click open in new tab, middle click open in background tab, that's mostly all I ever use. 

I should add I'm on Manjaro Linux, and another tab related add-on I use is "all-tabs-helper". Yes if you think that advanced option to ignore explicitly setting the parent tab id will help I'd be glad to test it, if by that you mean apply it to my current use habits. Beyond that I can follow instructions, not real tech savvy. 

There doesn't seem to be any rhyme or reason to the behavior that I can tell, it won't do it a few times then it happens again. Had a hunch it might be because of windows with a larger tab count but just tested on a 200+ window and a 65 tab window and they both did the same, kind of random behavior. 
thanks

**Comment by ssborbis** [comment]
I've added the option `disableNewTabSorting` to the advanced menu that will disable declaring which tab was the 'parent' allowing whatever the browser's default sorting is to take place. In your case, that should mean leaving it up to your tab managing addon.

You can sideload to test or wait on the next release

**Comment by ttggzz** [comment]
heh, well thanks for this but it appears I have wasted your time! I happened to go to the "open tabs next to current" ffx page and saw this, second paragraph at top: "After the addition of the browser.tabs.insertAfterCurrent setting in about:config you do not need this extension anymore." ! 
https://addons.mozilla.org/en-US/firefox/addon/open-tabs-next-to-current/?src=search
Removed OTNC, changed about:config browser.tabs.insertAfterCurrent setting to "true" and tested it out. A couple dozen searches with CS, no movement of tab position and the search opened next to the current tab. Sweet! Thanks again for your time and sorry about that, maybe this info will help someone else out. Will post back if anything changes but looks like good to go. Great add-on you have here!
